view/frontend/templates/js/replacejs.phtml

```diff
@@ -551,26 +554,15 @@ if (!$block->isOnPageFromWhiteList() && !$block->isMinicartEnabled()) { return;
             },
 
             check: function () {
-                /**
-                 * On Bolt button click check if guest checkout is allowed.
-                 * Display login popup to guest customers if it is not. The
-                 * Magento customerData and authenticationPopup objects are
-                 * used.
-                 */
-                // check if / wait until the observable objects are set.
-                // proceed, allow checkout if initial checkout flag is set
-                if (!initiateCheckout && !shoppingCart().data_id) return false;
-
-                // do not open checkout if the cart is empty
-                if (shoppingCart().summary_count === 0) return false;
-
-                // check if login is required
-                // skip the check if initial checkout flag is set
-                if (!initiateCheckout && !customer().firstname && !shoppingCart().isGuestCheckoutAllowed) {
-                    // if authentication is required for checkout set a cookie
-                    // for auto opening Bolt checkout after login
-                    setInitiateCheckoutCookie();
-                    authenticationPopup.showModal();
+                if (BoltState.checkResult === null) {
+                    popUpOpen = true;
+                    return BoltState.checkBarrier.promise;
+                }
+                if (!BoltState.checkResult) {
+                    if (!initiateCheckout && !BoltState.magentoCart.isGuestCheckoutAllowed && !BoltState.magentoCustomer.firstname ) {
```

Comment:
Why did the comment for this section disappear?
Could `initiateCheckout` be brought into your `BoltState` variable?